mindmeld/models/embedder_models.py

```diff
@@ -14,119 +14,738 @@
+                results.append(dict(zip(tgt_texts, similarity_scores)))
+            else:
+                if not _no_sort:  # sort results in descending scores
+                    n_scores = len(similarity_scores)
+                    if n_scores > top_n:
+                        top_inds = similarity_scores.argpartition(n_scores - top_n)[-top_n:]
+                        result = sorted(
+                            [(tgt_texts[ii], similarity_scores[ii]) for ii in top_inds],
+                            key=lambda x: x[1],
+                            reverse=True)
+                    else:
+                        result = sorted(zip(tgt_texts, similarity_scores),
+                                        key=lambda x: x[1],
+                                        reverse=True)
+                    results.append(result)
+                else:
+                    result = list(zip(tgt_texts, similarity_scores))
+                    results.append(result)
+
+        if is_single:
+            return results[0]
+
+        return results
 
 
 class BertEmbedder(Embedder):
     """
-    Encoder class for bert models as described here: https://github.com/UKPLab/sentence-transformers
+    Encoder class for bert models based on https://github.com/UKPLab/sentence-transformers
     """
 
     DEFAULT_BERT = "bert-base-nli-mean-tokens"
 
-    def load(self, **kwargs):
-        if self.model_name == "default":
-            bert_model_name = self.DEFAULT_BERT
-            logger.info("No bert model specifications passed, using default.")
+    # class variable to cache bert model(s);
+    #   helps to mitigate keeping duplicate sets of large weight matrices, especially
+    #   when creating mutliple BERT based Entity Resolvers
+    CACHE_MODELS = {}
+
+    def __init__(self, app_path, **kwargs):
+        """
+        Args:
+            batch_size (int): the batch size used for the computation
+            show_progress_bar (bool): Output a progress bar when encode sentences
+            output_value (str): Default sentence_embedding, to get sentence embeddings.
+                Can be set to token_embeddings to get wordpiece token embeddings.
+                Choices are `sentence_embedding` and `token_embedding`
+            convert_to_numpy (bool): If true, the output is a list of numpy vectors. Else, it is a
+                list of pytorch tensors.
+            convert_to_tensor (bool): If true, you get one large tensor as return. Overwrites any
+                setting from convert_to_numpy
+            device: Which torch.device to use for the computation
+            concat_last_n_layers (int): number of hidden outputs to concat starting from last layer
+            normalize_token_embs (bool): if the (sub-)token embs are to be individually normalized
+        """
+        super().__init__(app_path, **kwargs)
+
+        # init runtime configs for the model
+        self._batch_size = kwargs.get("batch_size", 16)
```

Comment:
Should we standardize the default `batch_size` across all mindmeld features? Currently the default `batch_size` for the paraphraser is 8.

Reply:
That's a good point. We should. Also, we should standardize the way a developer inputs/modifies batchsize as well as GPU usage. Can a developer do that as part of paraphraser config?

Reply:
Currently the user can only modify the batch size

Reply:
@vembar I modified it in this PR. Just for future reference, there is a `batch_size` param in lstm.py that uses a value of 20!